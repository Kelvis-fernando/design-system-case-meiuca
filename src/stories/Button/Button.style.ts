import tokens from "../../../tokens/global.json";

export const getButtonStyle = (
  primary: boolean,
  size: string
): React.CSSProperties => {
  const setSize = () => {
    switch (size) {
      case "small":
        return tokens.spacing.size.quarck.value;
      case "medium":
        return tokens.spacing.size.nano.value;
      case "large":
        return tokens.spacing.size.xxxs.value;
    }
  };
  return {
    color: primary
      ? tokens.neutral.color[1].value
      : tokens.neutral.color[5].value,
    backgroundColor: primary
      ? tokens.brand.color.primary[3].value
      : tokens.neutral.color[1].value,
    borderRadius: tokens.radius.size.sm.value,
    padding: setSize(),
    fontWeight: tokens.font.weight.bold.value,
    border: primary
      ? tokens.border.size.hairline.value
      : `solid ${tokens.border.size.hairline.value} ${tokens.neutral.color[5].value}`,
    cursor: "pointer",
    fontFamily: tokens.font.family.highlight.value,
  };
};
