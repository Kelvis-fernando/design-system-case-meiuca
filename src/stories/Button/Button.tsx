import React from "react";
import { buttonStyle } from "./Button.style";
import { ButtonProps } from "../../types/ButtonProps";

export const Button = ({
  primary = false,
  size = "medium",
  label,
  ...props
}: ButtonProps) => {
  return (
    <button type="button" style={buttonStyle(primary, size)} {...props}>
      {label}
    </button>
  );
};
