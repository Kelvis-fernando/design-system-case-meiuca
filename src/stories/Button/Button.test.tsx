import { render, screen } from "@testing-library/react";
import { Button } from "./Button";

test("renders learn react link", () => {
  const { container } = render(<Button label={"Button"} />);
  const linkElement = screen.getByText(/button/i);

  expect(container).toMatchSnapshot();
  expect(linkElement).toBeInTheDocument();
});
