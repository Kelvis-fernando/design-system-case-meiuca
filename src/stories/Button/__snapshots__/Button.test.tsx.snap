// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <button
    style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255); border-radius: 8px; padding: 8px; font-weight: 700; border: 1px solid #000000; cursor: pointer; font-family: 'Archivo';"
    type="button"
  >
    Button
  </button>
</div>
`;
